src/java/org/apache/cassandra/index/sai/disk/v1/V1OnDiskFormat.java

```diff
@@ -236,11 +248,14 @@ public Set<IndexComponent> perColumnIndexComponents(IndexContext indexContext)
     }
 
     @Override
-    public int openFilesPerSSTableIndex()
+    public int openFilesPerSSTableIndex(boolean hasClustering)
     {
-        // For the V1 format there are always 4 open files per SSTable - token values, primary key trie,
-        // primary key blocks, primary key block offsets
-        return 4;
+        // For the V1 format the number of open files depends on whether the table has clustering. For wide tables
+        // the number of open files will be 6 per SSTable - token values, partition sizes index, partition key blocks,
+        // partition key block offsets, clustering key blocks & clustering key block offsets and for skinny tables
+        // the number of files will be 4 per SSTable - token values, partition key sizes, partition key blocks &
+        // partition key block offsets.
+        return hasClustering ? 6 : 4;
```

Comment:
```suggestion
        return hasClustering ? WIDE_PER_SSTABLE_COMPONENTS.size() : SKINNY_PER_SSTABLE_COMPONENTS.size();
```

Reply:
Not directly related to the changes, but I have noticed that `IndexDescriptor#sizeOnDiskOfPerIndexComponents(IndexContext)` a few lines below is unused, we could remove it.

Reply:
It's used in `SSTableIndex.sizeOfPerColumnComponents`

Reply:
The problem with using the size of the component sets is that not all the components in the sets are opened. Some of them represent readers that use the opened components but have separate metadata.